.github/workflows/lint-and-test.yaml

```
@@ -2,43 +2,186 @@ name: Lint and test PR
 
 on: pull_request
 
+# Prevent to consume free CI minutes by not running two workflows in the CI at the same time.
+concurrency: 
+  group: pr-testing
+  cancel-in-progress: true
+
 jobs:
-  need-test:
+  paths-filter:
+    name: Filter which jobs need to be run
     runs-on: ubuntu-latest
     outputs:
-      charts: ${{ steps.list-changed.outputs.changed }}
+      workflows: ${{ steps.filter.outputs.workflows }}
+      build: ${{ steps.filter.outputs.build }}
+      charts: ${{ steps.filter.outputs.charts }}
     steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-      - uses: helm/chart-testing-action@v2.3.1
-      - id: list-changed
-        run: |
-          changed=$(ct --config .github/ct.yaml list-changed)
-          if [[ -n "$changed" ]]; then
-            echo "changed=true" >> $GITHUB_OUTPUT
-          fi
+    - uses: actions/checkout@v3
+    - uses: dorny/paths-filter@v2
+      id: filter
+      with:
+        filters: |
+          workflows:
+            - '.github/workflows/**'
+          build:
+            - 'build/**'
+          charts:
+            - 'charts/**'
 
-  lint-chart:
+
+  docker-build:
+    name: Build PowerDNS image
     runs-on: ubuntu-latest
-    needs: need-test
-    if: ${{ needs.need-test.outputs.charts == 'true' }}
+    needs: paths-filter
+    if: ${{ needs.paths-filter.outputs.build == 'true' || needs.paths-filter.outputs.workflows == 'true' }}
     steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-      - uses: helm/chart-testing-action@v2.3.1
-      - run: helm plugin install https://github.com/helm-unittest/helm-unittest
-      - uses: manusa/actions-setup-minikube@v2.7.2
-        with:
-          minikube version: v1.29.0
-          kubernetes version: v1.25.0
-          github token: ${{ secrets.GITHUB_TOKEN }}
-
-      - run: ct --config .github/ct.yaml lint --debug
+    - name: Checkout code
+      uses: actions/checkout@v3
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
 
-  docker-build:
+    - name: Build image
+      uses: docker/build-push-action@v4
+      with:
+        context: build
+        tags: ghcr.io/txqueuelen/stateless-dns:ci
+        outputs: type=docker,dest=/tmp/stateless-dns.tar
+
+    - name: Upload artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: docker-image
+        path: /tmp/stateless-dns.tar
+
+
+  lint-charts:
+    name: Lint charts
```

If we're running a big-ass E2E test here I'd suggest to rename this to something that sounds heavier.
```suggestion
    name: Lint and E2E test charts
```